services/src/datasets/external/sentinel_s2_l2a_cogs.rs

```diff
@@ -0,0 +1,640 @@
+
+        // request all features in zone in order to be able to determine the temporal validity of individual tile
+        let bbox =
+            SpatialReference::new(SpatialReferenceAuthority::Epsg, self.zone.epsg).area_of_use()?;
+
+        Ok(vec![
+            (
+                "collections[]".to_owned(),
+                "sentinel-s2-l2a-cogs".to_owned(),
+            ),
+            (
+                "bbox".to_owned(),
+                format!(
+                    "[{},{},{},{}]", // array-brackets are not used in standard but required here for unknkown reason
+                    bbox.lower_left().x,
+                    bbox.lower_left().y,
+                    bbox.upper_right().x,
+                    bbox.upper_right().y
+                ),
+            ), // TODO: order coordinates depending on projection
+            (
+                "datetime".to_owned(),
+                format!("{}/{}", t_start.to_rfc3339(), t_end.to_rfc3339()),
+            ),
+            ("limit".to_owned(), "500".to_owned()),
+        ])
+    }
+
+    async fn load_all_features<T: Serialize + ?Sized>(
+        &self,
+        params: &T,
+    ) -> Result<Vec<StacFeature>> {
+        let mut features = vec![];
+
+        let mut collection = self.load_collection(params, 1).await?;
+        features.append(&mut collection.features);
+
+        let num_pages =
+            (collection.context.matched as f64 / collection.context.limit as f64).ceil() as u32;
+
+        for page in 2..=num_pages {
+            let mut collection = self.load_collection(params, page).await?;
+            features.append(&mut collection.features);
+        }
+
+        Ok(features)
+    }
+
+    async fn load_collection<T: Serialize + ?Sized>(
+        &self,
+        params: &T,
+        page: u32,
+    ) -> Result<StacCollection> {
+        let client = reqwest::Client::new();
+        let text = client
+            .get(&self.api_url)
+            .query(&params)
+            .query(&[("page", &page.to_string())])
+            .send()
+            .await
+            .context(error::Reqwest)?
+            .text()
+            .await
+            .context(error::Reqwest)?;
+
+        serde_json::from_str(&text).map_err(|error| error::Error::StacJsonRespone {
+            url: self.api_url.clone(),
+            response: text,
+            error,
+        })
+    }
+
+    fn time_range_request(time: &TimeInterval) -> Result<(DateTime<Utc>, DateTime<Utc>)> {
+        let t_start =
+            time.start()
+                .as_utc_date_time()
+                .ok_or(geoengine_operators::error::Error::DataType {
+                    source: geoengine_datatypes::error::Error::NoDateTimeValid {
+                        time_instance: time.start(),
+                    },
+                })?;
+
+        // shift start by 1 day to ensure getting the most recent data for start time
+        let t_start = t_start - Duration::days(1);
```

Comment:
i guess we can also move it by -1 minutes. this will be enough to include a tile if it started before.

Reply:
are you sure? is the satellite also recording while it's on the night side?

Reply:
no but then there is no tile at night. i think tats a better semantic then a tile which is for ever valid

Reply:
if I only shift by 1 minute there may be no tile at all because the latest tile was from the prevous day. ThatÄs what I menat.

Reply:
yes but i think this is ok.  ypu need to aggregate anyway

Reply:
89397a1bc02a646b760c2b2655eec984131b749b